src/parser.c

```diff
@@ -0,0 +1,117 @@
+#include "parser.h"
+#include "util.h"
+#include <assert.h>
+#include <stdlib.h>
+
+/**
+ * @brief Initialize each parser according to the grammar
+ *
+ * @param psr
+ */
+void init_parser(struct parser *psr)
+{
+    psr->Ident = mpc_new("ident");
+    psr->Number = mpc_new("number");
+    psr->Character = mpc_new("character");
+    psr->String = mpc_new("string");
+    psr->Factor = mpc_new("factor");
+    psr->Term = mpc_new("term");
+    psr->Lexp = mpc_new("lexp");
+    psr->Stmt = mpc_new("stmt");
+    psr->Exp = mpc_new("exp");
+    psr->Typeident = mpc_new("typeident");
+    psr->Decls = mpc_new("decls");
+    psr->Args = mpc_new("args");
+    psr->Body = mpc_new("body");
+    psr->Procedure = mpc_new("procedure");
+    psr->Main = mpc_new("main");
```

Comment:
Since `main` doesn't occur in the grammar anymore, I think we should we get rid of the combinator as well.

Reply:
Nice catch. I am updating that.